cli_helpers/tabular_output/output_formatter.py

```diff
@@ -139,7 +143,12 @@ def format_output(self, data, headers, format_name=None,
         (_, _preprocessors, formatter,
          fkwargs) = self._output_formats[format_name]
         fkwargs.update(kwargs)
-        column_types = self._get_column_types(data)
+        if column_types is None:
+            data = list(data)
+            column_types = self._get_column_types(data)
+            data = iter(data)
```

Comment:
I hope this will prevent regressions, this will make code fail in both situations, when column_types is omitted and when it's specified, where for example, an iterator is required for preprocessors.

For example, I'm afraid we'll forget to test the situation where column_types aren't specified for a new formatter. If this new formatter isn't iterator proof it will fail when column_types are specified (by for example mycli).

Does this make sense? I'm open to suggestions, maybe we can make a pytest to run all formatters / preprocessors and which if they handle iterators well?